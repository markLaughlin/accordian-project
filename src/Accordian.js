import React from 'react'

class Accordian extends React.Component {
    static defaultProps = { sections: [] };

    state = {
        message: "This is the message"
    }

    handleButtonClick(index){
        console.log("A button was clicked");
        console.log({index});
        const currentMessage = this.props.sections[index].content;
        console.log({currentMessage});
        this.setState({message: currentMessage})
    }

    renderButtons(){
        const arrayOfLis = this.props.sections.map((item, index) =>
         <li key={index}>
             <button key={index} onClick={() => this.handleButtonClick(index)}>            
             {item.title}
            </button>
        </li>
        )

        return (
        <ul>
         {arrayOfLis}
        </ul>
         )
    }

    renderContent(){
        return (
        <div>
            {this.state.message}
        </div>
        )
    }

    render(){
        return(
        <div className = "mainDiv">
        {this.renderButtons()}
        {/* {this.renderContent()} */}
        {!!this.props.sections.length && this.renderContent()}
        </div>
        )
    }       
}

export default Accordian