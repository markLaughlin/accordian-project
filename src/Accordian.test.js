import React from 'react';
import ReactDOM from 'react-dom';
import Accordian from './Accordian';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`Accordian Component`, () => {

    const sections = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
      ]

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordian />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders empty given no sections', () => {
    const wrapper = shallow(<Accordian />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the first section by default', () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('closes the first section and opens any clicked section', () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
 
})