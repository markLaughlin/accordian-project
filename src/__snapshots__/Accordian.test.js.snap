// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component closes the first section and opens any clicked section 1`] = `
<div
  className="mainDiv"
>
  <ul>
    <li
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
  <div>
    Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
  </div>
</div>
`;

exports[`Accordian Component renders empty given no sections 1`] = `
<div
  className="mainDiv"
>
  <ul />
</div>
`;

exports[`Accordian Component renders the first section by default 1`] = `
<div
  className="mainDiv"
>
  <ul>
    <li
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
  <div>
    This is the message
  </div>
</div>
`;
